Indicators/SqueezeMomentum.cs

```diff
@@ -0,0 +1,109 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// The SqueezeMomentum indicator calculates whether the market is in a "squeeze" condition,
+    /// determined by comparing Bollinger Bands to Keltner Channels. When the Bollinger Bands are
+    /// inside the Keltner Channels, the indicator returns 1 (squeeze on). Otherwise, it returns -1 (squeeze off).
+    /// </summary>
+    public class SqueezeMomentum : BarIndicator, IIndicatorWarmUpPeriodProvider
+    {
+        /// <summary>
+        /// The Bollinger Bands indicator used to calculate the upper, lower, and middle bands.
+        /// </summary>
+        private readonly BollingerBands _bollingerBands;
+
+        /// <summary>
+        /// The Average True Range (ATR) indicator used to calculate the Keltner Channels.
+        /// </summary>
+        private readonly AverageTrueRange _averageTrueRange;
+
+        /// <summary>
+        /// The multiplier applied to the Average True Range for calculating Keltner Channels.
+        /// </summary>
+        private readonly decimal _keltnerMultiplier;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SqueezeMomentum"/> class.
+        /// </summary>
+        /// <param name="name">The name of the indicator.</param>
+        /// <param name="bollingerPeriod">The period used for the Bollinger Bands calculation.</param>
+        /// <param name="bollingerMultiplier">The multiplier for the Bollinger Bands width.</param>
+        /// <param name="keltnerPeriod">The period used for the Average True Range (ATR) calculation in Keltner Channels.</param>
+        /// <param name="keltnerMultiplier">The multiplier applied to the ATR for calculating Keltner Channels.</param>
+        public SqueezeMomentum(string name, int bollingerPeriod, decimal bollingerMultiplier, int keltnerPeriod, decimal keltnerMultiplier) : base(name)
+        {
+            _bollingerBands = new BollingerBands(bollingerPeriod, bollingerMultiplier);
+            _averageTrueRange = new AverageTrueRange(keltnerPeriod, MovingAverageType.Simple);
```

Comment:
Would be nice for the user to be able to select the moving average type, as an argument to the indicator. Maybe optional with `Simple` as default